sdk/core/core-amqp/src/retry.ts

```diff
@@ -118,11 +138,28 @@ async function checkNetworkConnection(host: string): Promise<boolean> {
  * The retries when made are done so linearly on the given operation for a specified number of times,
  * with a specified delay in between each retry.
  *
- * @param {RetryConfig<T>} config Parameters to configure retry operation.
+ * @param {RetryConfig<T>} config Parameters that define what type of retry will be performed
  *
  * @return {Promise<T>} Promise<T>.
  */
 export async function retry<T>(config: RetryConfig<T>): Promise<T> {
+  if (config.exponentialRetry === true) {
+    return exponentialRetry(config);
+  } else {
+    return linearRetry(config);
```

Comment:
At a quick glance below are the differences between the `linearRetry` and `exponentialRetry` functions that have been introduced based on current implementation
- Way to figure out the delay between retries
- Default values being set for different properties of the `config` 
- Way to calculate `totalNumberOfAttempts`
     - This should not differ between the two. It should be `maxRetries` + 1 

Given such small differences between the two, I don't see the merits of splitting the `retry` method into two, resulting in much more lines of code copied between the two. 

Please consider going back to the single `retry` method, apply defaults to all properties of the config, and have a single `if (config.exponentialRetry)/else` block to determine the delay between retries